staff/javier-sevilla/precourse/pasapalabra-html/pasapalabra.js

```diff
@@ -0,0 +1,268 @@
+      user = prompt('INTRODUZCA EL USUARIO');
+      if (user != "" && user != null) {
+         noFinUser = false;      
+      }
+    } while (noFinUser);        
+};
+
+function mostrarRanking() {
+   let i, j;
+   let stringRanking = "";
+   userRanking.sort(function(a, b){return b.points - a.points});
+   for (i = 0; i < userRanking.length; i++) {
+       j = i + 1;
+       if (userRanking.length > 9) {
+           i = 99999;
+       }
+       stringRanking += j + "- " + userRanking[i].name + 
+                        " Puntos: " + userRanking[i].points + " <br>";  
+    }
+    document.getElementById("rank").innerHTML = "<b>RANKING: </b> " + "<br>" + stringRanking;
+};
+
+function addRanking(user, points) {
+    let ranking;
+    ranking = { name: user,
+        points: points
+      };
+      userRanking.push(ranking);
+};
+ 
+function inicializaStatus() {
+    let i;
+    for (i = 0; i < preguntas.length; i++) {
+       preguntas[i].status = 0;
+     }
+};
+ 
+function numRandom(num) {
+    let numRan = 0;
+    numRan = Math.floor(Math.random() * num);
+    if (numRan === 0) {
+       numRan = numRandom(num);
+    }
+    return numRan;
+};
+ 
+function asignarPreguntas() {
+     randomNumber = numRandom(4);
+     switch (randomNumber) {
+       case 1: 
+             preguntas = questions;
+             break;
+       case 2:
+             preguntas = questions1;
+             break;
+       case 3:
+             preguntas = questions2;
+             break;  
+     }
+};
+
+function mostrarPrimeraPregunta() {    
+     indicePreguntas = 0;
+     document.getElementById("preguntas").innerHTML = preguntas[indicePreguntas].question;
+};
+
+function MostrarPregunta() {    
+    document.getElementById("preguntas").innerHTML = preguntas[indicePreguntas].question;
+};
+
+function addRanking(user, points) {
+    let ranking;
+    ranking = { name: user,
+        points: points
+      };
+      userRanking.push(ranking);
+ };
+
+function submit() {   
+    let respuesta = document.getElementById("frm1").value;
+    let i;
+
+    respuesta = respuesta.toLocaleLowerCase();
+
+
+    if (respuesta != preguntas[indicePreguntas].answer) {
+        totalFails += 1;
+        preguntas[indicePreguntas].status = 99;
+        document.getElementById(indicePreguntas).style.background = "rgb(173, 14, 9)";
+     } else {
+        totalPoints += 1;
+        preguntas[indicePreguntas].status = 1;
+        document.getElementById(indicePreguntas).style.background = "rgb(127, 204, 56)";
+        
+    
+     };
+
+     document.getElementById("frm1").value = "";
+
+     if (indicePreguntas > 25) {
+        indicePreguntas = 0; 
+     }
+
+
+     for (i = indicePreguntas; i < preguntas.length ;i++) {
+        if (preguntas[i].status === 0) {
+            indicePreguntas = i;
+            i = 9999;
+        };
+      };
+
+    for (i = 0; i < preguntas.length ;i++) {
+        if (preguntas[i].status === 0) {
+            noFin = true;
+            i = 9999;
```

Comment:
As a suggestion I would recommend a unified variable for testing reasons in those iterators equal to i=9999, by this way you would be able to change all iterators at the same time. var1=9999 and in each iterator i =var1.

Reply:
i finally  change the code i=9999 to i=Number.MAX_SAFE_INTEGER;